tests/kubevirt_helper.go

```diff
@@ -0,0 +1,200 @@
+			log.InfoD("Storage class of PVC attached to VM [%s] in namespace [%s] is [%s]", v.Name, v.Namespace, storageClass)
+
+			// Add the disks to the VM
+			pvcs, err := CreatePVCsForVM(v, 1, storageClass, size)
+
+			if err != nil {
+				return false, err
+			}
+
+			specListInterfaces := make([]interface{}, len(pvcs))
+			for i, pvc := range pvcs {
+				// Converting each PVC to interface for appending to SpecList
+				specListInterfaces[i] = pvc
+			}
+			appCtx.App.SpecList = append(appCtx.App.SpecList, specListInterfaces...)
+
+			err = AddPVCsToVirtualMachine(v, pvcs)
+			if err != nil {
+				return false, err
+			}
+
+			err = RestartKubevirtVM(v.Name, v.Namespace, true)
+			if err != nil {
+				return false, err
+			}
+			log.InfoD("Sleep for 5mins for vm to come up")
+			time.Sleep(5 * time.Minute)
+
+			//After adding the pvcs check the number of disks in the VM
+			vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{appCtx})
+			if err != nil {
+				return false, err
+			}
+			for _, v := range vms {
+				t = func() (interface{}, bool, error) {
+					diskCountOutput, err := GetNumberOfDisksInVM(v)
+					if err != nil {
+						return nil, false, fmt.Errorf("failed to get number of disks in VM [%s] in namespace [%s]", v.Name, v.Namespace)
+					}
+					// Total disks will be numberOfVolumes plus the container disk
+					if diskCountOutput != numberOfDisks+diskCount {
+						return nil, true, fmt.Errorf("expected number of disks in VM [%s] in namespace [%s] is [%d] but got [%d]", v.Name, v.Namespace, numberOfDisks+1, diskCountOutput)
+					}
+					return diskCountOutput, false, nil
+				}
+				d, err = task.DoRetryWithTimeout(t, 10*time.Minute, 30*time.Second)
+				if err != nil {
+					return false, err
+				}
+				if diskCount == d.(int) {
+					return false, fmt.Errorf("number of disks in VM [%s] in namespace [%s] is same as before adding disks", v.Name, v.Namespace)
+				}
+				diskCount = d.(int)
+				log.InfoD("Number of disks in VM [%s] in namespace [%s] is [%d]", v.Name, v.Namespace, diskCount)
+
+			}
+		}
+	}
+	return true, nil
+}
+
+// GetStorageClassOfVmPVC returns the storage class of pvc attached to the VM
+func GetStorageClassOfVmPVC(vm *scheduler.Context) (string, error) {
+	// Get the PVC object from the VM
+	nameSpace := vm.App.NameSpace
+	pvcs, err := core.Instance().GetPersistentVolumeClaims(nameSpace, nil)
+	if err != nil {
+		return "", err
+	}
+
+	// Get the PVCs attached to the VM
+	for _, pvc := range pvcs.Items {
+		ScName, err := core.Instance().GetStorageClassForPVC(&pvc)
+		if err != nil {
+			return "", err
+		}
+		return ScName.Name, nil
+	}
+	return "", fmt.Errorf("failed to get storage class of PVC attached to VM [%s] in namespace [%s]", vm.App.Key, vm.App.NameSpace)
+}
+
+// startAndWaitForVMIMigration starts the VM migration and waits for the VM to be in running state in the new node
+func StartAndWaitForVMIMigration(virtualMachine *scheduler.Context, ctx context1.Context) (bool, error) {
+
```

Comment:
@spai-px : I see @pureneelesh reviewed this method already in the other PR. Please incorporate those comments here. 
Also I think the other PR is now no longer required right ?

Reply:
Ok will do.